File: bench/Reference/sequential_async.java
public class sequential_async {
    static int slow(int x) throws InterruptedException { Thread.sleep(x); return x; }
    public static void main(String[] args) throws Exception {
        int a = slow(100);
        int b = slow(a);
        int c = slow(b);
        int d = slow(c);
        System.out.println(d);
    }
}

File: bench/Reference/sequential_async.py
import asyncio


async def slow(x):
    await asyncio.sleep(x / 1000)
    return x


async def main():
    a = await slow(100)
    b = await slow(a)
    c = await slow(b)
    d = await slow(c)
    print(d)


asyncio.run(main())
